### PR TITLE
337 stop n

### PR DESCRIPTION
**Pull request by thanasions**

### CODE REVIEW

File: tests/test_nmf_wrapper.py [hunk 1]
```diff
@@ -21,31 +21,31 @@ def setUpClass(cls):
         cls.__nmf = nmf_topic_modelling(nmf_topics, tfidf_obj.tfidf_matrix)
```
Comment (IanGrimstead) [inline tests/test_nmf_wrapper.py]: Index scores have changed - i.e. [0][3302] became [0][1399] - why the change?

File: tests/test_text_processing.py [hunk 2]
```diff
@@ -103,113 +118,215 @@ def test_WordAnalyser_ngrams(self):
...
-                           'field region',
-                           'layer thickness',
-                           'layer thickness',
-                           'layer thickness',
-                           'layer thickness',
-                           'conductive material cover',
-                           'material cover field',
-                           'cover field region',
-                           'field region adjacent']
+        expected_ngrams = [
```
Comment (IanGrimstead) [inline tests/test_text_processing.py]: Wow - many more ngrams, is that expected?

File: tests/test_tfidf_mask.py [hunk 3]
```diff
@@ -61,30 +61,61 @@ def init_mask(self, cpc, min_n, uni_factor=0.8):
 
     def test_num_non_zeros_no_clean_rows(self):
         self.init_mask('Y02', 2)
-        self.assertEqual(2024, len(self.__tfidf_mask.data))
+        self.assertEqual(6917, len(self.__tfidf_mask.data))
```
Comment (IanGrimstead) [inline tests/test_tfidf_mask.py]: Again, many more ngrams - is that expected?

File: tests/test_filter_terms.py [hunk 4]
```diff
@@ -19,8 +19,10 @@ def setUp(self):
 
     def test_embeddings_filter_binary(self):
         user_queries = ['pharmacy', 'health', 'chemist']
-        weights_vec_expected = [1.0,
-                                1.0,
+        weights_vec_expected = [0.0,
```
[the remark this answers is not in the document]
Comment (thanasions) [reply tests/test_filter_terms.py]: no match :)

File: tests/test_nmf_wrapper.py (hunk 1 above)
Comment (thanasions) [reply tests/test_nmf_wrapper.py]: cause it is very sparse. you get a lot of zeros....

File: tests/test_text_processing.py (hunk 2 above)
Comment (thanasions) [reply tests/test_text_processing.py]: yes, with the recent changes. Our code works for >1000 docs.

File: tests/test_tfidf_mask.py (hunk 3 above)
Comment (thanasions) [reply tests/test_tfidf_mask.py]: yes, i will explain at some point.

File: scripts/pipeline.py [hunk 5]
```diff
@@ -45,11 +46,10 @@ def __init__(self, data_filename, docs_mask_dict, pick_method='sum', ngram_range
             dataframe = data_factory.get(data_filename)
             utils.checkdf(dataframe, calculate_timeseries, docs_mask_dict, text_header)
             utils.remove_empty_documents(dataframe, text_header)
-
             self.__tfidf_obj = tfidf_from_text(text_series=dataframe[text_header],
                                                ngram_range=ngram_range,
                                                max_document_frequency=max_df,
-                                               tokenizer=LemmaTokenizer())
+                                               tokenizer=LemmaTokenizer(), min_df=floor(log10(dataframe.shape[0])))
```
[the remark this answers is not in the document]
Comment (thanasions) [reply scripts/pipeline.py]: just came up with it...:)

**Comment by IanGrimstead** [comment]
If the test `test_embeddings_filter_binary` returns all zeros, it's a bogus test alas as its missed the data (allegedly) - can you mod the terms so we get at least something non-zero? Otherwise it'll be very easy to accidentally pass that test...
